clang/test/CXX/drs/cwg24xx.cpp

```diff
@@ -45,6 +45,34 @@ void fallthrough(int n) {
 #endif
 }
 
+namespace cwg2428 { // cwg2428: 19
+#if __cplusplus >= 202002L
+template <typename>
+concept C [[deprecated]] = true; // #C
```

Comment:
I'd like an additional test for rejecting:
```
template <typename>
[[deprecated]] concept C = true;
```
which isn't allowed per the grammar (though I find this slightly surprising and it may be an oversight from WG21).

It would also be good to have a test explicitly showing that we reject non-C++-style attributes.

Reply:
I added test with alternative attribute placing.

> It would also be good to have a test explicitly showing that we reject non-C++-style attributes.

It's not possible to write such a test, because we accept `deprecated` in more spellings than just C++11 one.

Reply:
`concept C __attribute__((deprecated)) = true; ` should be ill-formed. that should be easy to test

Reply:
As it turns out, GCC does support GNU-style attributes https://compiler-explorer.com/z/dccfdofKf
We should probably follow suite. Sorry for the confusion Vlad
(Which means you need to add tests + support for that)

Reply:
Non-standard spellings of this attribute are out of scope of the Core issue, so I'd like to left them for a subsequent PR.

Reply:
I added support for GNU and MSVC spellings.